src/prefect/deployments.py

```diff
@@ -555,13 +560,21 @@ async def build_from_flow(
             deployment.version = flow.version
         if not deployment.description:
             deployment.description = flow.description
-        if not deployment.storage:
+
+        if not deployment.storage and not is_docker_based:
             deployment.path = str(Path(".").absolute())
+        elif not deployment.storage and is_docker_based:
+            # only update if a path is not already set
+            if not deployment.path:
+                deployment.path = "/opt/prefect/deployment"
```

Comment:
Note we also use `/opt/prefect` during the image build process. This doesn't collide, but we may want to be careful.

Reply:
Yup good call - I did choose it because it was our common place to put things in arbitrary docker, but to your point we should be aware of what we do with that directory